backend/layers/persistence/persistence.py

```diff
@@ -393,6 +422,13 @@ def delete_canonical_collection(self, collection_id: CollectionId) -> None:
             canonical_collection = session.query(CollectionTable).filter_by(id=collection_id.id).one_or_none()
             if canonical_collection:
                 canonical_collection.tombstone = True
+            # Tombstone the Datasets individually as well; technically not necessary but will protect us from bugs
```

Comment:
The comment is misleading. I should get rid of that. The tombstoning of the constituent Datasets is necessary to prevent errant individual Dataset access. Otherwise...
1) We would have to check whether a Dataset is tombstoned AND whether its parent Collection is tombstoned whenever Dataset access is happening; this could require extra queries in some places
2) Examining the raw data in the db becomes more difficult; there would be inconsistencies in how Datasets would be marked as tombstoned (some with a tombstoned parent Collection, some with their own `tombstone` column marked as true)